### PR TITLE
improve automated movement

### PR DESCRIPTION
**Pull request by owalk**

### CODE REVIEW

File: game/entities/entity.py [hunk 1]
```diff
@@ -109,41 +207,175 @@ def set_image_rotation(self, image):
 
         return pygame.transform.rotate(image, angle)
 
-    def set_image_direction(self, image):
-        """Set a new image to the correct cardinal direction.
+    def move_based_on_state(self):
```
Comment (Sean-Nishi) [inline game/entities/entity.py:210]: Line 73 has 'Flee' as a state but there is no mention of Flee in this move_based_on_state() method. We could add a stub method to resolve this.

File: game/entities/entity.py [hunk 2]
```diff
@@ -25,7 +30,25 @@ class Entity(Tile, ABC):
...
+        Create sprite sheet from path
+    import_assets(self)
+        import images into smaller images
+    animate(self)
+        animation loop for entity
+    set_status_by_current_rotation(self)
+        set status per current direction
+    get_angle_from_direction(self, axis: str)
+        get the angle for sprite rotation per compass direction
+    set_bad_sprites(self, bad_sprites: pygame.sprite.Group)
```
Comment (Sean-Nishi) [inline game/entities/entity.py:43]: Lines 43-50 uses 'sets' or and line 51 uses 'Handles' which is imperative voice and capitalized

File: game/entities/skeleton.py [hunk 3]
```diff
@@ -1,91 +1,66 @@
...
 
 class Skeleton(Entity):
     """First enemy class. Called Skeleton.
 
     Inherits from Entity. Move method is overwritten using logic described
     in the docs. Still uses Entity's collision_check method.
+
+    Attributes
+    ----------
+    sprite_sheet : SpriteSheet
```
Comment (Sean-Nishi) [inline game/entities/skeleton.py:15]: Lines 16-26 'holds' is imperative voice

**Review by Sean-Nishi — CHANGES_REQUESTED** [review]
Imperative voice and maybe adding a Flee stub method

File: game/entities/entity.py (hunk 2 above)
Comment (owalk) [reply game/entities/entity.py:43]: Fixed, also the capitalization was wrong on many of these lines. Fixed that as well

File: game/entities/entity.py (hunk 1 above)
Comment (owalk) [reply game/entities/entity.py:210]: Will add in another commit

File: game/entities/skeleton.py (hunk 3 above)
Comment (owalk) [reply game/entities/skeleton.py:15]: fix, and fixed capitalization

File: game/entities/entity.py [hunk 4]
```diff
@@ -25,7 +30,25 @@ class Entity(Tile, ABC):
 
     Methods
     -------
-    collision_check(self, direction)
+    set_sprite_sheet(self, path: str)
+        Create sprite sheet from path
```
Comment (Sean-Nishi) [inline game/entities/entity.py:34]: Capitalization for 'Create'

File: game/entities/damsel.py [hunk 5]
```diff
@@ -44,42 +29,30 @@ def __init__(self, pos, groups, obstacle_sprites):
...
+        obstacle_sprites : list of sprite groups
+            which sprites in the level damsel cannot walk through
         """
         super().__init__(groups)
 
         damselMovementImagePath = "graphics/damsel/damselWalking.png"
-        self.damselAnimations = SpriteSheet(damselMovementImagePath)
-        damselSelfImageRect = pygame.Rect(0, 0, SPRITE_WIDTH, SPRITE_HEIGHT)
-        self.image = self.damselAnimations.image_at(damselSelfImageRect)
+        self.sprite_sheet = self.get_sprite_sheet(damselMovementImagePath)
```
Comment (Sean-Nishi) [inline game/entities/damsel.py:42]: Vars are camel case, methods are snake case

File: game/entities/damsel.py [hunk 6]
```diff
@@ -44,42 +29,30 @@ def __init__(self, pos, groups, obstacle_sprites):
...
+            0, 0, settings.ENTITY_WIDTH, settings.ENTITY_HEIGHT
+        )
+        self.image = self.sprite_sheet.image_at(damselSelfImageRect)
         self.setColorKeyWhite()
         self.rect = self.image.get_rect(topleft=pos)
         # modify model rect to be a slightly less tall hitbox.
         self.hitbox = self.rect.inflate(0, -10)
         random.seed(time.time())
         self.timer = 100
         self.obstacle_sprites = obstacle_sprites
```
Comment (Sean-Nishi) [inline game/entities/damsel.py:53]: Vars are camel case. Change to 'self.obstacleSprites'

File: game/entities/damsel.py [hunk 7]
```diff
@@ -44,42 +29,30 @@ def __init__(self, pos, groups, obstacle_sprites):
...
-        walkingRightRect = (0, SPRITE_HEIGHT * 2, SPRITE_WIDTH, SPRITE_HEIGHT)
-
-        # animation states in dictionary
-        self.animations = {
-            "up": self.damselAnimations.load_strip(walkingUpRect, 3),
-            "down": self.damselAnimations.load_strip(walkingDownRect, 3),
-            "left": self.damselAnimations.load_strip(walkingLeftRect, 3),
-            "right": self.damselAnimations.load_strip(walkingRightRect, 3),
-        }
+        self.enemy_sprites = pygame.sprite.Group()
```
Comment (Sean-Nishi) [inline game/entities/damsel.py:54]: Camel case

File: game/entities/entity.py [hunk 8]
```diff
@@ -25,7 +30,25 @@ class Entity(Tile, ABC):
...
+    set_bad_sprites(self, bad_sprites: pygame.sprite.Group)
+        sets the bad sprites group
+    set_good_sprites(self, good_sprites: pygame.sprite.Group)
+        sets the good sprites group
+    set_image_rotation(self, image: pygame.Surface) -> pygame.Surface
+        Rotate image per compass direction
+    set_player(self, player: pygame.Sprite)
+        sets the player
+    collision_handler(self)
         Handles the collision check for entities
```
Comment (Sean-Nishi) [inline game/entities/entity.py:52]: Capitalization

File: game/entities/entity.py [hunk 9]
```diff
@@ -43,20 +66,75 @@ def __init__(self, groups):
         # skeletons
         self.bad_sprites = pygame.sprite.Group()
```
Comment (Sean-Nishi) [inline game/entities/entity.py:67]: Camel case

File: game/entities/entity.py [hunk 10]
```diff
@@ -43,20 +66,75 @@ def __init__(self, groups):
...
 
-    def animate(self):
+        # init empty player
+        self.player = pygame.sprite.Sprite()
+
+        # setting up state machine
+        self.states = Enum("skeleton_state", ["Patrol", "Attack", "Flee", "Follow"])
+        self.current_state = self.states.Patrol
```
Comment (Sean-Nishi) [inline game/entities/entity.py:74]: Camel case

File: game/entities/entity.py [hunk 11]
```diff
@@ -43,20 +66,75 @@ def __init__(self, groups):
...
+
+        # the skeleton is attacking
+        self.sprite_to_attack = pygame.sprite.Sprite()
```
Comment (Sean-Nishi) [inline game/entities/entity.py:77]: Camel

File: game/entities/entity.py [hunk 12]
```diff
@@ -43,20 +66,75 @@ def __init__(self, groups):
...
+
+        # for automated movements, store a previous timestamp
+        self.last_time_stored = 0
```
Comment (Sean-Nishi) [inline game/entities/entity.py:80]: Camel

**Review by Sean-Nishi — CHANGES_REQUESTED** [review]
Mostly comment capitalization

**Review by Sean-Nishi — APPROVED** [review]

**Pull request merged**